### PR TITLE
Fix CCE and NPE in client resource access action node context

### LINKED ISSUE #37100: CCE in implement method code action
**Description:**
Consider the following screen capture.
Peek 2022-07-22 11-09

stack trace
```
Error - 11:09:18 am] CodeAction 'ImplementMethodCodeAction' failed! {, error: 'class java.lang.String cannot be cast to class io.ballerina.compiler.api.symbols.TypeSymbol (java.lang.String is in module java.base of loader 'bootstrap'; io.ballerina.compiler.api.symbols.TypeSymbol is in unnamed module of loader 'app')'}
java.lang.ClassCastException: class java.lang.String cannot be cast to class io.ballerina.compiler.api.symbols.TypeSymbol (java.lang.String is in module java.base of loader 'bootstrap'; io.ballerina.compiler.api.symbols.TypeSymbol is in unnamed module of loader 'app')
	at org.ballerinalang.langserver.codeaction.providers.AbstractImplementMethodCodeAction.getDiagBasedTextEdits(AbstractImplementMethodCodeAction.java:93)
	at org.ballerinalang.langserver.codeaction.providers.ImplementMethodCodeAction.getCodeActions(ImplementMethodCodeAction.java:58)
	at org.ballerinalang.langserver.codeaction.CodeActionRouter.lambda$getAvailableCodeActions$4(CodeActionRouter.java:123)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
	at org.ballerinalang.langserver.codeaction.CodeActionRouter.lambda$getAvailableCodeActions$5(CodeActionRouter.java:113)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:177)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1655)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:497)
	at org.ballerinalang.langserver.codeaction.CodeActionRouter.getAvailableCodeActions(CodeActionRouter.java:110)
	at org.ballerinalang.langserver.codeaction.BallerinaCodeActionExtension.execute(BallerinaCodeActionExtension.java:46)
	at org.ballerinalang.langserver.codeaction.BallerinaCodeActionExtension.execute(BallerinaCodeActionExtension.java:34)
	at org.ballerinalang.langserver.LangExtensionDelegator.codeActions(LangExtensionDelegator.java:167)
	at org.ballerinalang.langserver.BallerinaTextDocumentService.lambda$codeAction$8(BallerinaTextDocumentService.java:332)
	at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:642)
	at java.base/java.util.concurrent.CompletableFuture$Completion.exec(CompletableFuture.java:479)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)


```

**Steps to reproduce:**
```Ballerina
public client class Client {
    public string url;

    public function init(string url) {
        self.url = url;
    }

    resource function post path2/[string id]/path4(string str, int a) returns string {
        return "";
    }

}


public function test() {
    Client cl = new ("");
    string a  = cl-> /path2/[""]/path4.post()
}

```
**Affected Versions:**
2201.2.x
**OS, DB, other environment details and versions:**

**Related Issues (optional):**

**Suggested Labels (optional):**

**Suggested Assignees (optional):**

### PR DESCRIPTION
**Pull request by malinthar**
## Purpose
$subject

Fixes #37100 

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Review by IMS94 — APPROVED** [review]

File: language-server/modules/langserver-core/src/test/resources/codeaction/implement-functions/config/implementMethodNegative1.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+    "line": 16,
+    "character": 25
+  },
+  "source": "implementMethodNegative1.bal",
+  "expected": [
+    {
+      "title": "Implement method"
+    }
+  ]
+}
```
Comment (mindula) [inline language-server/modules/langserver-core/src/test/resources/codeaction/implement-functions/config/implementMethodNegative1.json:12]: Let's add a new line here

File: language-server/modules/langserver-core/src/test/resources/hover/source/hover_for_api_docs_negative.bal [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+        return "";
+    }
+
+}
+
+
+public function test() {
+    Client cl = new ("");
+    string a  = cl-> /path2/[""]/path4.post()
+}
```
Comment (mindula) [inline language-server/modules/langserver-core/src/test/resources/hover/source/hover_for_api_docs_negative.bal:18]: Add a new line

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/hover/HoverUtil.java [hunk 3]
```diff
@@ -99,7 +99,7 @@ public static Hover getHover(HoverContext context) {
             HoverSymbolResolver symbolResolver =
                     new HoverSymbolResolver(context, semanticModel.get());
             Optional<Symbol> symbol = context.getNodeAtCursor().apply(symbolResolver);
-            if (!symbolResolver.isSymbolReferable()) {
+            if (symbol == null || symbol.isEmpty() || !symbolResolver.isSymbolReferable()) {
```
[the remark this answers is not in the document]
Comment (malinthar) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/hover/HoverUtil.java:102]: When the ExternalTreeList node is visited `null` is returned

[the remark this answers is not in the document]
Comment (IMS94) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/hover/HoverUtil.java:102]: I see. +1

**Review by mohanvive — DISMISSED** [review]

**Review by mohanvive — APPROVED** [review]

**Pull request merged**